app/models/miq_server/monitor/system_limits.rb

```diff
@@ -0,0 +1,138 @@
+require 'miq-system'
+
+module MiqServer::WorkerManagement::Base::Monitor::SystemLimits
```

Comment:
NOTE I think most of this can actually move into the Process subclass, just have to deal with https://github.com/ManageIQ/manageiq/pull/21445/files#diff-2186c0d28d483db3064af2d8cfe25bf0d9f1a40779d537626eb9522a48210537R11